service/api/service.go

```diff
@@ -156,15 +156,30 @@ func (s *serviceImpl) ApiV1WorkflowWaitForStateCompletion(
 ) (wresp *iwfidl.WorkflowWaitForStateCompletionResponse, retError *errors.ErrorAndStatus) {
 	defer func() { log.CapturePanic(recover(), s.logger, &retError) }()
 
-	var workflowId string
+	var parentWfId, currentWorkflowId, _ string
+
+	response, err := s.client.DescribeWorkflowExecution(ctx, req.GetWorkflowId(), "", nil)
+	if err != nil {
+		return nil, s.handleError(err, WorkflowWaitForStateCompletionApiPath, req.WorkflowId)
+	}
+
+	if response.FirstRunId == "" {
+		parentWfId = req.WorkflowId // Cadence
+	} else {
+		parentWfId = response.FirstRunId // Temporal
```

Comment:
very nit: parentId? it's not workflowId Lol.

Reply:
👍 done